### PR TITLE
mplayer: unbundle `ffmpeg`

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by cho-m** [comment]
May need to backport some fixes to get newer `ffmpeg` though not sure how many changes are needed.

Debian is using SVN revision 38408 (w/ FFmpeg 5) and Arch is using 38412 (w/ FFmpeg 6).

File: Formula/mplayer.rb [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+  stable do
+    url "https://mplayerhq.hu/MPlayer/releases/MPlayer-1.5.tar.xz"
+    sha256 "650cd55bb3cb44c9b39ce36dac488428559799c5f18d16d98edb2b7256cbbf85"
+
+    # Backports to fix build with FFmpeg 6:
+    # - r38361 (configure, av_helpers.c: Fix compilation against latest FFmpeg) to fix:
+    #   av_helpers.c:57:34: error: no member named 'decode' in 'struct AVCodec'
+    # - r38411 (ad_spdif.c: Use proper way to set avformat option) to fix:
+    #   libmpcodecs/ad_spdif.c:115:57: error: no member named 'priv_data_size' in 'struct AVOutputFormat'
+    patch :p0, :DATA
```
Comment (cho-m) [inline Formula/mplayer.rb:17]: Added bare minimum backports to fix compilation locally, but there are some other commits related to "latest FFmpeg".

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
Still draft?

**Comment by cho-m** [comment]
May switch the patches to Git mirror (https://git.mplayerhq.hu/gitweb/mplayer.git) before merge.

Also, I may need to do some local testing to make sure there isn't any obvious issues.

**Comment by cho-m** [comment]
Sadly hit some audio issue with FFmpeg 6.0 (ffvorbis codec) on both HEAD and patched stable. Haven't had chance to debug further.

Linking to FFmpeg 5.1 worked. Other audio format like opus worked.

**Comment by carlocab** [comment]
Let's just try using the preferred `ffmpeg` version here without patching. We can add this to the versioned dependencies conflict allowlist if needed.

As it is, with the bundled ffmpeg, it already loads different versions of `ffmpeg` into memory (one version is our own, another version is the bundled one) and that doesn't seem to have caused problems.

Let's use the latest `ffmpeg` for `head`, though.

**Pull request closed without merging**